File: src/集训/week3-2/C.cpp
#include <bits/stdc++.h>

using namespace std;

const int INF = 0x3f3f3f3f;
const int maxn = 310;

int g[maxn][maxn];

int main() {
    int n, m, T;
    cin >> n >> m >> T;
    memset(g, 0x3f, sizeof(g));
    for (int i = 1; i <= m; i++) {
        int u, v, w;
        cin >> u >> v >> w;
        g[u][v] = w;
    }

    for (int k = 1; k <= n; k++)
        for (int i = 1; i <= n; i++)
            for (int j = 1; j <= n; j++)
                g[i][j] = min(g[i][j], max(g[i][k], g[k][j]));
    while (T--) {
        int a,b;
        cin >> a >> b;
        if(g[a][b] == INF) {
            cout << -1 << endl;
        }else{
            cout << g[a][b] << endl;
        }
    }
}

File: src/集训/week3-2/C.py
n, m, t = map(int, input().split())

w = [[]]
inf = 100000000
for i in range(1, n + 1):
    w.append([inf])
    for j in range(1, n + 1):
        w[i].append(inf)

for i in range(m):
    a, b, c = map(int, input().split())
    w[a][b] = c

for k in range(1, n + 1):
    for i in range(1, n + 1):
        for j in range(1, n + 1):
            w[i][j] = min(w[i][j], max(w[i][k], w[k][j]))
for i in range(t):
    a, b = map(int, input().split())
    if(w[a][b] == inf):
        print(-1)
    else:
        print(w[a][b])
